### DISCUSSION TITLE
ICYStream and WiFiManager - Initial WiFi credentials unknown

### DISCUSSION THREAD

**Discussion by Ephesian** [pschatzmann/arduino-audio-driver#38] [Q&A] [answered]
Hello

I am a Noob and trying to expand my knowledge, so apologies
Can someone please point me in the direction. I am trying to stream MP3 from a url, and I have managed to get the example to run if I explicitly set my WiFi .

What I would prefer to do is use WiFiManager (https://github.com/tzapu/WiFiManager) to find and setup my WiFi.  The issue that I have come across is that because I do not the ssid or password I am unable to set const char* ssid or const char* pwd.  If I set ssid equal to const char* ssid = WiFi.SSID().

Error output:-
src/main.cpp:20:25: error: cannot resolve overloaded function 'SSID' based on conversion to type 'const char*'
 const char* ssid = WiFi.SSID();
                         ^~~~
*** [.pio\build\esp-wrover-kit\src\main.cpp.o] Error 1

if I cast the WiFi.SSID() using c_str()
 and ICYStream url(ssid, pwd);
 the code reports [   775][E][WiFiSTA.cpp:232] begin(): SSID too long or missing!

any help welcome

**Comment by pschatzmann** [comment, marked as answer]
You're thinkng way too complicated: the ssid and password are optional parameters that you can provide if you did not connect to the WiFi otherwise...

Just check the class documentation in the AudioTools!

ps. use the AudioTools project! This is just for the drivers if you use e.g. an AudioKit with an ES8388 codec chip...
